packages/agents/cartographer/poller/src/lib/operations/messages.ts

```diff
@@ -56,43 +57,48 @@ export const updateMessages = async () => {
   for (const originDomain of domains) {
     for (const destinationDomain of domains) {
       if (originDomain == destinationDomain) continue;
-      const offset = 0;
+      let offset = 0;
       const limit = 100;
-
-      logger.debug("Updating messages", requestContext, methodContext, {
-        originDomain,
-        destinationDomain,
-        offset,
-        limit,
-      });
-      const pendingMessages = await database.getUnProcessedMessagesByDomains(
-        originDomain,
-        destinationDomain,
-        limit,
-        offset,
-      );
-      if (pendingMessages.length > 0) {
-        logger.debug("Pending messages", requestContext, methodContext, {
+      let done = false;
+      const xMessages: XMessage[] = [];
+      while (!done) {
```

Comment:
Let's run this a max of like 5 times per iteration instead of running it forever. We want to rely on the lambda retrying this instead of looping through everything here.